src/providers/live/live-provider.ts

```diff
@@ -1,3 +1,222 @@
+    super(player, eventManager, logger, types);
+    this._pushNotification = new PushNotificationPrivider(this._player, this._logger);
+    this._initNotification();
+    this._pushNotification.registerToPushServer(this._player.getMediaInfo().entryId, types, this._handleConnection, this._handleConnectionError);
+    this._addBindings();
+  }
+
+  private _onTimedMetadataLoaded = ({payload}: any): void => {
+    // TODO: handle dash format
+    const id3TagCues = payload.label === ID3_TAG_LABEL ? payload.cues : [];
+    if (id3TagCues.length) {
+      try {
+        const id3Timestamp = Math.ceil(JSON.parse(id3TagCues[id3TagCues.length - 1].value.data).timestamp / 1000);
+        this._logger.debug(`Received new id3 timestamp: ${id3Timestamp}`);
+        if (id3Timestamp) {
+          this._id3Timestamp = id3Timestamp;
+        }
+      } catch (e) {
+        this._logger.debug('Failed retrieving id3 tag metadata');
+      }
+    }
+  };
+
+  private _onTimeUpdate = (): void => {
+    // TODO: handle dash format
+    const newTime = Math.ceil(this._player.currentTime);
+    if (newTime === this._currentTime) {
+      return;
+    }
+    this._currentTime = newTime;
+    if (this._seekDifference !== null && this._currentTimeLive) {
+      // update _currentTimeLive after seek
+      this._currentTimeLive = this._currentTimeLive - this._seekDifference;
+    } else if (this._id3Timestamp) {
+      if (this._id3Timestamp === this._currentTimeLive) {
+        // prevent updating if calculated _currentTimeLive value the same as _id3Timestamp
+        this._id3Timestamp = null;
+        return;
+      }
+      // update _currentTimeLive from id3Tag time
+      this._currentTimeLive = this._id3Timestamp;
+    } else {
+      // update _currentTimeLive between id3Tags
+      this._currentTimeLive++;
+    }
+
+    this._id3Timestamp = null;
+    this._seekDifference = null;
+  };
+
+  private _handleSeeking = (): void => {
+    this._seekDifference = Math.ceil(this._currentTime - this._player.currentTime);
+  };
+
+  private _addBindings() {
+    this._eventManager.listen(this._player, this._player.Event.TIMED_METADATA, this._onTimedMetadataLoaded);
+    this._eventManager.listen(this._player, this._player.Event.SEEKING, this._handleSeeking);
+    this._eventManager.listen(this._player, this._player.Event.TIME_UPDATE, this._onTimeUpdate);
+  }
+
+  private _handleConnection = () => {
+    this._logger.debug('Connected to push server');
+  };
+
+  private _handleConnectionError = () => {
+    this._logger.error('Got an error during connection to push server');
+  };
+
+  private _initNotification(): void {
+    this._pushNotification.init();
+    this._constructPushNotificationListener();
+  }
+
+  private _prepareCuePoints<T extends ThumbPushNotificationData | SlideViewChangePushNotificationData>(cuePoint: T, cb: (cuePoint: T) => void) {
+    if (!this._player.currentTime) {
```

Comment:
whats is the usecase here?

Reply:
data from push notification might come before first id3 tag comes. The solution propose wait till id3 tag comes and `this._player.currentTime` calculated

Reply:
done